src/cert_store.rs

```diff
@@ -447,8 +582,62 @@ mod test {
             .unwrap();
     }
 
+    fn find_existing_cert_helper(pfx: &[u8], pass: &str) { 
+        // import provided pfx file to the memory store, key will be persisted since Cryptography::NO_PERSIST_KEY is not set
+        let memory_store = PfxImportOptions::new()
+            .include_extended_properties(true)
+            .password(pass)
+            .import(pfx)
+            .unwrap();
+        
+        // set identity to the cert that matches the key inside the pfx file
+        let mut identity = None;
+        for cert in memory_store.certs() {
+            if cert
+                .private_key()
+                .silent(true)
+                .compare_key(true)
+                .acquire()
+                .is_ok()
+            {
+                identity = Some(cert);
+                break;
+            }
+        }
+        let identity = identity.unwrap();
+
+        // open temporary "TestRustMy" store for current user and add identity if it is not already in the store
+        let mut store = CertStore::open_current_user("TestRustMy").unwrap();
+        store.add_cert(&identity, CertAdd::Always).unwrap();
+
+        let result = CertStore::find_existing_cert(&mut store, &identity).unwrap();
+
+        assert_eq!(result, identity);
```

Comment:
>result, identity [](http://example.com/codeflow?start=19&length=16)

This isn't doing pointer compare:

impl PartialEq for CertContext {
    fn eq(&self, other: &CertContext) -> bool {
        self.get_encoded_bytes() == other.get_encoded_bytes()
    }